### PR TITLE
display alert bar

### LINKED ISSUE #294: [web] Create alert banner
Alert creation UI was created in PR #292.
Storage/types/API are forthcoming, but expect an endpoint that will give you an array of current alerts (not sure how we're going to handle multiples -- maybe just work on displaying one for now).

Expect a `TAlert` to look something like:
```
export interface TAlert {
  _id: string;
  createdAt: Date;
  createdBy: TUser;
  color: string; // Hex string, eg #FF11BB
  detail: string | null;
  end: Date;
  icon: string; // fontawesome icon identifier; ex: "fas fa-snowflake"
  isCancelled: boolean;
  lastModifiedAt: Date;
  lastModifiedBy: TUser;
  start: Date;
  title: string;
}
```

Here is a pseudo-mockup of the alert:
Screen Shot 2020-03-11 at 7 02 37 PM

### PR DESCRIPTION
**Pull request by jacobvenable**
This PR closes #294

## What does this PR do?
- adds an alert bar to the top of public-facing app
- adds corresponding drawer to display more details on the current alert(s)
- creates `<MuiMarkdown/>` component that wraps markdown, converted by [react-markdown](https://github.com/rexxars/react-markdown), in MUI components
- adds font awesome to the project to display alert icons chosen by user in admin interface

### Alert Bar

_alert bar with one alert_
Screen Shot 2020-03-27 at 7 37 19 AM

_alert bar with multiple alerts_
Screen Shot 2020-03-27 at 7 34 25 AM

### Alert Drawer

_alert drawer with multiple alerts_
Screen Shot 2020-03-27 at 7 34 47 AM

_alert without alert details on hover/touch_
Screen Shot 2020-03-27 at 7 35 13 AM

_alert using markdown_
Screen Shot 2020-03-27 at 7 34 59 AM

## How does this PR make you feel? [:link:](http://giphy.com/categories/emotions/)

you've been warned

### CODE REVIEW

File: packages/upswyng-web/src/components/AlertPanel.tsx [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+import Grid from "@material-ui/core/Grid";
+import MuiMarkdown from "./MuiMarkdown";
+import React from "react";
+import { TAlertFull } from "@upswyng/upswyng-types";
+import { Theme } from "@material-ui/core/styles/createMuiTheme";
+import Tooltip from "@material-ui/core/Tooltip";
+import Typography from "@material-ui/core/Typography";
+import makeStyles from "@material-ui/styles/makeStyles";
+
+// const AlertPanelTooltip = ({children}:{})
```
Comment (rhinodavid) [inline packages/upswyng-web/src/components/AlertPanel.tsx]: missed that guy

File: packages/upswyng-web/src/components/AlertPanel.tsx [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+  expansionPanelDetails: {
+    display: "block",
+  },
+  expansionPanelSummary: (alert: TAlertFull) => ({
+    cursor: !alert.detail ? "default" : undefined,
+  }),
+}));
+
+interface Props extends ExpansionPanelProps {
+  alert: TAlertFull;
```
Comment (rhinodavid) [inline packages/upswyng-web/src/components/AlertPanel.tsx]: the users will only get a `TAlert` (`TAlertFull` is used for the admin interface)

File: packages/upswyng-web/src/components/AlertPanel.tsx (hunk 1 above)
Comment (jacobvenable) [reply packages/upswyng-web/src/components/AlertPanel.tsx]: oh derp. [upderped](https://github.com/CodeForBoulder/upswyng/pull/309/commits/cd1925966e22eb8b36c1c0c9c302d99d295e8cc3)

File: packages/upswyng-web/src/components/AlertPanel.tsx (hunk 2 above)
Comment (jacobvenable) [reply packages/upswyng-web/src/components/AlertPanel.tsx]: ah gotcha 👍 [updated](https://github.com/CodeForBoulder/upswyng/pull/309/commits/b20d6bb918c879294228f88709235728dc52acfe)

**Pull request merged**